### PR TITLE
[mlir][vector] Refactor vector-transfer-unroll.mlir (NFC)

### PR DESCRIPTION
**Pull request by pabloantoniom**
Following @banach-space initiative of refactoring vector tests, this patch refactors `vector-transfer-unroll.mlir`:

- Unified identation of `FileCheck` commands
- Separated tests individually (previously there was a big test containing a couple of tests)
- All memref input arguments are re-named to %mem.
- All vector input arguments are re-named to %vec.
- LIT variables are update to be consistent with input arguments.
- Renamed all output arguments as %res.

### CODE REVIEW

File: mlir/test/Dialect/Vector/vector-transfer-unroll.mlir [hunk 1]
```diff
@@ -2,363 +2,373 @@
...
-//  ORDER-NEXT:   %[[PASS5:.*]] = vector.extract_strided_slice %[[ARG3]] {offsets = [4, 2], sizes = [2, 2], strides = [1, 1]} : vector<6x4xf32> to vector<2x2xf32>
-//  ORDER-NEXT:   %[[VGT5:.*]] = vector.gather {{.*}}[%[[C0]], %[[C0]], %[[C0]]] [%[[IDX5]]], %[[MASK5]], %[[PASS5]] : memref<?x?x?xf32>, vector<2x2xindex>, vector<2x2xi1>, vector<2x2xf32> into vector<2x2xf32>
-//  ORDER-NEXT:   %[[VEC5:.*]] = vector.insert_strided_slice %[[VGT5]], %[[VEC4]] {offsets = [4, 2], strides = [1, 1]} : vector<2x2xf32> into vector<6x4xf32>
-//  ORDER-NEXT:   return %[[VEC5]] : vector<6x4xf32>
-
-func.func @vector_gather_unroll(%arg0 : memref<?x?x?xf32>,
+// ORDER-SAME:    %[[MEM:.*]]: memref<?x?x?xf32>
+// ORDER-SAME:    %[[ARG1:.*]]: vector<6x4xindex>
+// ORDER-SAME:    %[[ARG2:.*]]: vector<6x4xi1>
+// ORDER-SAME:    %[[ARG3:.*]]: vector<6x4xf32>
```
Comment (banach-space) [inline mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:333]: How about more descriptive var names, e.g. `INDICES`, `MASK`, `PT`?

File: mlir/test/Dialect/Vector/vector-transfer-unroll.mlir [hunk 2]
```diff
@@ -2,363 +2,373 @@
 // RUN: mlir-opt %s -test-vector-transfer-unrolling-patterns=reverse-unroll-order --split-input-file | FileCheck %s --check-prefix=ORDER
 
 // CHECK-LABEL: func @transfer_read_unroll
```
Comment (banach-space) [inline mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:4]: CHECK-LABEL and CHECK-SAME seem to be repeated in this file quite a lot. Consider using a "shared" label like here:
* https://github.com/llvm/llvm-project/blob/main/mlir/test/Dialect/Vector/linearize.mlir

File: mlir/test/Dialect/Vector/vector-transfer-unroll.mlir
Comment (banach-space) [inline mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:1]: Is this testing a specific pattern? If yes, could you add a comment similar to e.g.:
* https://github.com/llvm/llvm-project/blob/a50b9633357007ff886f3fd228ca4b8a9b9b9852/mlir/test/Dialect/Vector/vector-transfer-collapse-inner-most-dims.mlir#L3-L6

If this is exercising multiple patterns, I'd probably skip.

**Review by banach-space — COMMENTED** [review]
Thank you for helping out with this 😅 

Already looks great, but I've made a few small suggestions that could make this even greater 🙏🏻 WDYT?

File: mlir/test/Dialect/Vector/vector-transfer-unroll.mlir (hunk 1 above)
Comment (pabloantoniom) [reply mlir/test/Dialect/Vector/vector-transfer-unroll.mlir]: Good catch! Fixed 👍

File: mlir/test/Dialect/Vector/vector-transfer-unroll.mlir
Comment (pabloantoniom) [reply mlir/test/Dialect/Vector/vector-transfer-unroll.mlir:1]: It is using multiple patterns (at least `UnrollTransferReadPattern`, `UnrollTransferWritePattern` and `UnrollGatherPattern`) and the tests are mixed (not grouped like in the example you show) so I think it's better to skip it for this case.

**Comment by pabloantoniom** [comment]
> Thank you for helping out with this 😅
> 
> Already looks great, but I've made a few small suggestions that could make this even greater 🙏🏻 WDYT?

Looks good! I've addressed all your comments and also fixed a small detail I didn't notice before. In `transfer_read_unroll_tensor` I wrongly replaced the name of the argument with `%mem` (but it shouldn't be, because it's a tensor).

**Review by banach-space — APPROVED** [review]
LGTM, thank you for addressing my comments  and for improving this test 🙏🏻 

Do you have commit access? If not, can land this for you.

**Comment by pabloantoniom** [comment]
> LGTM, thank you for addressing my comments and for improving this test 🙏🏻
> 
> Do you have commit access? If not, can land this for you.

Yes I have. Thank you for the review! 👍

**Pull request merged**